### PR TITLE
ENH: replace depcrecated distutils use in convenience function

### PR DESCRIPTION
**Pull request by mattip**
Related to #18588, removes a use of distutils.

xref @nilskattenbeck-bosch who asked about this use.

### CODE REVIEW

File: numpy/ctypeslib.py [hunk 1]
```diff
@@ -120,26 +120,23 @@ def load_library(libname, loader_path):
             If there is no library with the expected extension, or the
             library is defective and cannot be loaded.
         """
-        if ctypes.__version__ < '1.0.1':
-            import warnings
-            warnings.warn("All features of ctypes interface may not work "
-                          "with ctypes < 1.0.1", stacklevel=2)
-
```
Comment (mattip) [inline numpy/ctypeslib.py:127]: This part is no longer relevant, ctypes.__version__ is 1.1.0

File: numpy/ctypeslib.py [hunk 2]
```diff
@@ -120,26 +120,23 @@ def load_library(libname, loader_path):
...
             # Try to load library with platform-specific name, otherwise
-            # default to libname.[so|pyd].  Sometimes, these files are built
+            # default to libname.[so|dll|dylib].  Sometimes, these files are built
             # erroneously on non-linux platforms.
-            from numpy.distutils.misc_util import get_shared_lib_extension
+            base_ext = ".so"
+            if sys.platform.startswith("darwin"):
+                base_ext = ".dylib"
+            elif sys.platform.startswith("win"):
+                base_ext = ".dll"
```
Comment (mattip) [inline numpy/ctypeslib.py:136]: This is the first part of `get_shared_lib_extension()`

File: numpy/ctypeslib.py [hunk 3]
```diff
@@ -120,26 +120,23 @@ def load_library(libname, loader_path):
...
             so_ext = get_shared_lib_extension()
-            libname_ext = [libname + so_ext]
-            # mac, windows and linux >= py3.2 shared library and loadable
-            # module have different extensions so try both
-            so_ext2 = get_shared_lib_extension(is_python_ext=True)
+            libname_ext = [libname + base_ext]
+            so_ext2 = sysconfig.get_config_var("EXT_SUFFIX")
```
Comment (mattip) [inline numpy/ctypeslib.py:139]: This is equivalent to the code it replaces

**Review by seberg — APPROVED** [review]
Thanks, LGTM.  I guess it might end up with a slightly cleaner version and with trying `.so` in odd platforms.  I doubt this matters, I think this is easier to grok then the old code anyway.

**Pull request merged**